scipy/optimize/tests/test__shgo.py

```diff
@@ -27,7 +27,7 @@ def __init__(self, bounds, expected_x, expected_fun=None,
 def wrap_constraints(g):
     cons = []
     if g is not None:
-        if (type(g) is not tuple) and (type(g) is not list):
+        if not isinstance(g, tuple | list):
```

Comment:
```suggestion
        if not isinstance(g, (tuple, list)):
```

Reply:
Indeed, [union types should work as well](https://docs.python.org/3/library/functions.html#isinstance), but only in **Python ≥ 3.10**:
> If _classinfo is_ a tuple of type objects (or recursively, other such tuples) or a [Union Type](https://docs.python.org/3/library/stdtypes.html#types-union) of multiple types, return `True` if _object_ is an instance of any of the types.

Reply:
See: https://numpy.org/neps/nep-0029-deprecation_policy.html for python version roadmap